packages/host/app/resources/file.ts

```diff
@@ -181,7 +181,9 @@ class _FileResource extends Resource<Args> {
       size,
       url: response.url,
       write(content: string, flushLoader?: true) {
-        self.writing = self.writeTask.perform(this, content, flushLoader);
+        self.writing = self.writeTask
+          .linked() // If the parent task is cancelled, this task should be cancelled too. Helps mitigating flakiness in tests
```

Comment:
There was a warning emitted from ember-concurrency that either `linked()` or `unlinked()` should be added to child tasks to know what to do when the parent task gets cancelled (there is another task that is calling this one)

Reply:
it should be unlinked--if the component is destroyed we don't want the write to not happen

Reply:
maybe that's the issue with the flakiness?

Reply:
Yeah actually you made a good point - I have a hunch that occasionally we get into a weird situation where the modal is destroyed before the write is done causing the both tasks to be canceled (?)

Adding `unlinked()` helped. You are right we should let the child task run even if the parent task is canceled